### DISCUSSION TITLE
reportUnnecessaryTypeIgnoreComment but only for mypy comments?

### DISCUSSION THREAD

**Discussion by FrancescElies** [microsoft/pyright#4532] [Q&A] [answered]
We have a relative big codebase (70K lines) and currently we are evaluating mypy and pyright simultaneously.

I would like pyright to warn me about unused ignores, the following option does the job.
>  reportUnnecessaryTypeIgnoreComment [boolean or string, optional]: Generate or suppress diagnostics for a '# type: ignore' or '# pyright: ignore' comment that would have no effect if removed. The default value for this setting is 'none'.

The problem is that it complains about mypy specific stuff too, is there a way to instruct pyright to warn us only about unused `# pyright: ignore` comments only?

Thanks in advance for your advice

**Comment by erictraut** [comment, marked as answer]
The `reportUnnecessaryTypeIgnoreComment` check covers both `# type: ignore` and `# pyright: ignore`. It doesn't distinguish between the two.

The `# type: ignore` comment is common to all type checkers, not just mypy. It's documented in PEP 484. Pyright added `# pyright: ignore` as a pyright-specific facility. I think the best answer is for mypy to add a mypy-specific facility as well (e.g. `# mypy: ignore`). If you agree, I recommend upvoting [this issue in the mypy tracker](https://github.com/python/mypy/issues/12358).

**Reply by FrancescElies** [reply]
I agree, done, thanks once again
